### PR TITLE
Compile the library in a standard way

### PR DESCRIPTION
**Pull request by jspricke**

### CODE REVIEW

File: libreflexxestype2/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,6 @@
 cmake_minimum_required(VERSION 3.0.2)
 project(libreflexxestype2)
 
-# Compile as C++14, supported in ROS Melodic and newer
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED YES)
```
Comment (ChrisThrasher) [inline libreflexxestype2/CMakeLists.txt:6]: If we still want to set the language standard, which we should, lets use `target_compile_features` to do so. 

```cmake
target_compile_features(reflexxes PUBLIC cxx_std_14)
```

Doing it this way ensures that language requirements are transitively passed on to downstream targets.

Comment (jspricke) [reply libreflexxestype2/CMakeLists.txt:6]: you should not set language standards, the compiler has a sane default.

Comment (ChrisThrasher) [reply libreflexxestype2/CMakeLists.txt:6]: You should absolutely set the language requirements. If the compiler default matches what you require then the default is used. If a compiler supports C++14 but defaults to C++11 then we want to make sure the default isn't used. It's widely regarded as CMake best practices to specify language standard requirements.

File: libreflexxestype2/CMakeLists.txt [hunk 2]
```diff
@@ -17,44 +13,47 @@ include_directories(
...
-
-if(SUPPORTED MATCHES "true")
-  # Set location of build output
-  set(libreflexxes_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/Linux/x64/release/lib/shared/libReflexxesTypeII.so)
-
-  # Build Reflexxes library with a Make file
-  add_custom_command(
-    OUTPUT ${libreflexxes_LOCATION}
-    COMMAND make -C "${CMAKE_CURRENT_SOURCE_DIR}/Linux" clean64 all64 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-  )
```
Comment (tylerjw) [inline libreflexxestype2/CMakeLists.txt:40]: Is there a makefile we should also delete?  this is really weird that a c++ library would be built with cmake and they add a custom command to call make.

**Review by tylerjw — APPROVED** [review]

**Review by AndyZe — APPROVED** [review]
LGTM! Would you mind opening a PR to do the same on the Foxy branch?

**Pull request merged**